Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -870,11 +879,19 @@ void AppendPipeline::Track::initializeElements(AppendPipeline* appendPipeline, G
     appsinkDataEnteringPadProbeInformation.probeId = gst_pad_add_probe(appsinkPad.get(), GST_PAD_PROBE_TYPE_BUFFER, reinterpret_cast<GstPadProbeCallback>(appendPipelinePadProbeDebugInformation), &appsinkDataEnteringPadProbeInformation, nullptr);
 #endif
 
-#if ENABLE(ENCRYPTED_MEDIA)
     appsinkPadEventProbeInformation.appendPipeline = appendPipeline;
     appsinkPadEventProbeInformation.description = "appsink event probe";
-    appsinkPadEventProbeInformation.probeId = gst_pad_add_probe(appsinkPad.get(), GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM, reinterpret_cast<GstPadProbeCallback>(appendPipelineAppsinkPadEventProbe), &appsinkPadEventProbeInformation, nullptr);
-#endif
+    appsinkPadEventProbeInformation.probeId = gst_pad_add_probe(appsinkPad.get(), GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM,
+        [](GstPad* pad, GstPadProbeInfo* padProbeInfo, void* userData) {
+        ASSERT(GST_PAD_PROBE_INFO_TYPE(padProbeInfo) & GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM);
+        GstEvent* event = gst_pad_probe_info_get_event(padProbeInfo);
+        GST_DEBUG_OBJECT(pad, "Handling event %" GST_PTR_FORMAT " on append pipeline appsinkPad", event);
+        struct PadProbeInformation* padProbeInformation = static_cast<PadProbeInformation*>(userData);
+        AppendPipeline* appendPipeline = padProbeInformation->appendPipeline;
+        if (!appendPipeline)
+            return GST_PAD_PROBE_OK;
```

Comment:
Same as previous comment. This check was done in the old static probe. I can remove it, since apparently the probe is also removed in the AppendPipeline destructor and the null pointer situation shouldn't happen.